src/library.js

```diff
@@ -2016,6 +2016,9 @@ addToLibrary({
     throw 'unwind';
   },
 
+#if !MINIMAL_RUNTIME
+  _emscripten_runtime_keepalive_clear__deps: ['$runtimeKeepaliveCounter'],
```

Comment:
Two things:
* Perhaps `$runtimeKeepaliveCounter` should be removed below from the direct deps of `emscripten_force_exit`, since it will already get it implicitly from depending on this function.
* I'm not 100% sure about this, but could we instead have an alternative version of `_emscripten_runtime_keepalive_clear` that just doesn't do anything and doesn't depend on `runtimeKeepaliveCounter` at all, since in that case the variable isn't otherwise used? It doesn't seem to be incremented/decremented or checked in the case I was building. Not a huge deal either way, but if the other code paths avoid using the variable when in this mode, perhaps this one could avoid using the variable as well?

Reply:
First part done, good idea. 

Second part is harder.  Its something I've done for `noExitRuntime` but this stuff is pretty tricky to get right so I'll leave that for a followup.